src/agent/main.cpp

```diff
@@ -248,7 +261,16 @@ int main(int argc, char *argv[])
 #if OPENTHREAD_ENABLE_COVERAGE
         __gcov_flush();
 #endif
-        execvp(argv[0], argv);
+        std::vector<std::string> args(argv, argv + argc);
+        args.erase(std::remove_if(args.begin(), args.end(),
+                                  [](const std::string &arg) { return arg.rfind("--auto-attach", 0) == 0; }),
+                   args.end());
+        args.push_back("--auto-attach=0");
+        std::vector<char *> args_cstr(args.size() + 1);
+        std::transform(args.begin(), args.end(), args_cstr.begin(),
+                       [](const std::string &arg) { return const_cast<char *>(arg.c_str()); });
+        *args_cstr.rbegin() = nullptr;
```

Comment:
nit: may add a function to encapsulate the logic?

Reply:
Done.